crates/cxx-qt-gen/src/parser/qnamespace.rs

```diff
@@ -0,0 +1,107 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use syn::{ItemMacro, LitStr, Result};
+
+use crate::syntax::attribute::attribute_take_path;
+
+pub struct ParsedQNamespace {
+    /// The name of the namespace
+    pub namespace: String,
+    /// whether qml_element was specified
+    pub qml_element: bool,
+}
+
+impl ParsedQNamespace {
+    pub fn parse(mut mac: ItemMacro) -> Result<Self> {
+        let namespace: LitStr = syn::parse2(mac.mac.tokens)?;
+        let namespace = namespace.value();
+
+        let qml_element = attribute_take_path(&mut mac.attrs, &["qml_element"]).is_some();
+
+        if let Some(attr) = mac.attrs.first() {
+            return Err(syn::Error::new_spanned(
+                attr,
+                "qnamespace! macro must not have any attributes other than qml_element!",
+            ));
+        }
+
+        if let Some(ident) = mac.ident {
+            return Err(syn::Error::new_spanned(
+                ident,
+                "qnamespace! macro must not have an additional identifier",
+            ));
+        }
+
+        Ok(Self {
+            namespace,
+            qml_element,
+        })
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use std::assert_eq;
```

Comment:
why does this need to be included ? (and in `crates/cxx-qt-gen/src/generator/cpp/constructor.rs` in `main`) as elsewhere it doesn't need to be included ?

Reply:
it does, in fact, not need to be included :sweat_smile: